### PR TITLE
Prepare docs

### LINKED ISSUE #3: Transform inline-documentation to numpydoc style
To automatically render and generate a documentation out of the python package, the docstrings should follow some standards. I strongly recommend the [numpy style](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard). 

https://github.com/ClimateClara/ARC3O/blob/891ed42f5b4b5a24cafcaac6475e1998bcb21b9b/profile_functions.py#L50-L71

for instance then look like

```python
def build_temp_profile(surf_temp_ice,empty_temp_prof,winter_mask,fyi_mask,myi_mask,layer_amount,snow_opt,snd,sit,e_bias_fyi,e_bias_myi):
    """Build the temperature profile.

    This function builds linear temperature profiles over depth
    between the snow surface temperature and ice surface temperature
    and between  ice surface temperature and freezing temperature (-1.8°C)
	
    Parameters
    ----------
    surf_temp_ice: xarray.DataArray
        sea-ice (or snow) surface temperature in K
    empty_temp_prof: xarray.DataArray
        empty xarray with the expected dimensions of the output
    winter_mask: xarray.DataArray
        mask telling us which points are in "winter"
    layer_amount: int
        number of layers the profile should have
    snow_opt: str 
        ``'yes'`` if there is snow on top, ``'no'`` if there is no snow on top
    snd: xarray.DataArray
        snow thickness in m
    sit: xarray.DataArray
        sea-ice thickness in m
    e_bias_fyi, e_bias_myi: xarray.DataArray
        tuning parameter to correct for MEMLS bias in emissivity
	
    Returns
    -------
    xarray.DataArray
        tuned temperature profile over layer_amount in K

    Notes
    -----
    All input data arrays must be broadcastable towards each other!
    """

```

### PR DESCRIPTION
**Pull request by ClimateClara**
I have gone through all the steps in Issue #3 and #4 , closes #3  and #4

### CODE REVIEW

**Comment by Chilipp** [comment]
Hey @ClimateClara! Congratulations on the paper =) I'll have a look at this here asap.

But already now: I highly recomment that you add a .gitignore file to ignore the `_build`, `__pycache__` and `.pyc` files, etc. Luckily Github already provides good templates for that! Just click `Add File`, type `.gitignore` and it will provide you some templates (see the screencast below)

add_gitignore

File: readthedocs.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+conda:
+    file: docs/environment.yml
```
Comment (Chilipp) [inline readthedocs.yml]: this file is missing

File: setup.py [hunk 2]
```diff
@@ -56,19 +55,6 @@
...
-          'xarray',
-          'pandas',
-          'datetime',
-          'time',
-          'timeit',
-          'itertools',
-          'tqdm',
-          'subprocess',
-          'pathos.multiprocessing'
-      ],
```
Comment (Chilipp) [inline setup.py]: why did you remove the install requirements?

File: docs/start/installation.rst [hunk 3]
```diff
@@ -0,0 +1,9 @@
+.. _installation:
+
+How to install
+==============
+
+Currently, the easiest is to clone the `ARC3O git hub repository <github.com/ClimateClara/arc3o>`_.
```
Comment (Chilipp) [inline docs/start/installation.rst]:
```suggestion
Currently, the easiest is to clone the `ARC3O github repository <https://github.com/ClimateClara/arc3o>`__.
```

File: docs/start/installation.rst [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+
+This package is programmed in python 3.6 and should be working with all `python
+versions > 3.3`. Additional requirements are numpy, xarray, pandas, tqdm and pathos.
```
Comment (Chilipp) [inline docs/start/installation.rst]:
```suggestion
This package is programmed in python 3.6 and should be working with all `python
versions > 3.6`. Additional requirements are numpy, xarray, pandas, tqdm and pathos.
```
nobody supports python less than 3.5, and I strongly recommend to restrict yourself to versions that you used by yourself.

File: docs/start/installation.rst (hunk 3 above)
Comment (Chilipp) [inline docs/start/installation.rst]: probably this file is currently just placeholder? Because you should provide instructions that you can install it via 

```bash
git clone https://github.com/ClimateClara/arc3o.git
pip install arc3o/
```

or via

```bash
pip install git+https://github.com/ClimateClara/arc3o.git
```

File: docs/guide/howtorun.rst [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+Input data
+----------
+
+Currently, ARC3O is tailored for model output from ECHAM, the atmospheric module of the Max Planck Institute Earth System Model.
+However, it can be used for other models if the variable names are changed accordingly either in the ARC3O functions or the model's output.
+
+The input data for ARC3O should be divided into monthly files, found in the folder ``inputpath``.
+Also, to prepare the seasons and ice types mask, ARC3O needs one file in which all the data is merged, found in the folder ``inputpath0``.
+
+These monthly and overview files should contain fields [lat,lon,time] of the following variables:
```
Comment (Chilipp) [inline docs/guide/howtorun.rst]:
```suggestion
These monthly and overview files should contain fields [time,lat,lon] of the following variables:
```
or is the time dimension the last? would be quite unusual

File: docs/guide/howtorun.rst [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+Output
+------
+
+The output of ARC3O is written into several netcdf files to ``outputpath``:
+    * 'period_masks_assim.nc': Masks for ice type and seasons.
+    * 'profiles_for_memls_snowno_yyyymm.nc': Snow-free profiles of ice and snow properties.
+    * 'profiles_for_memls_snowyes_yyyymm.nc': Snow-covered profiles of ice and snow properties.
+    * 'TB_assim_yyyymm_f.nc': Ice surface brightness temperatures (H and V polarization) for grid cells with ice in cold conditions.
+    * 'TBtot_assim_yyyymm_f.nc': Brightness temperatures (H and V polarization) at the top of atmosphere (incl. other seasons than cold conditions and ocean and atmosphere contribution) for all ocean grid cells.
```
Comment (Chilipp) [inline docs/guide/howtorun.rst]:
```suggestion
    * ``'period_masks_assim.nc'``: Masks for ice type and seasons.
    * ``'profiles_for_memls_snowno_yyyymm.nc'``: Snow-free profiles of ice and snow properties.
    * ``'profiles_for_memls_snowyes_yyyymm.nc'``: Snow-covered profiles of ice and snow properties.
    * ``'TB_assim_yyyymm_f.nc'``: Ice surface brightness temperatures (H and V polarization) for grid cells with ice in cold conditions.
    * ``'TBtot_assim_yyyymm_f.nc'``: Brightness temperatures (H and V polarization) at the top of atmosphere (incl. other seasons than cold conditions and ocean and atmosphere contribution) for all ocean grid cells.
```

File: docs/conf.py [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+	'sphinxcontrib.bibtex'
+]
+
+# Add any paths that contain templates here, relative to this directory.
+templates_path = ['_templates']
+
+# List of patterns, relative to source directory, that match files and
+# directories to ignore when looking for source files.
+# This pattern also affects html_static_path and html_extra_path.
+exclude_patterns = ['_build', 'Thumbs.db', '.DS_Store']
```
Comment (Chilipp) [inline docs/conf.py]:
```suggestion
exclude_patterns = ['_build', 'Thumbs.db', '.DS_Store', 'guide/step?.rst']
```
otherwise step1.rst will be available via `/guide/step1.html` and via `guide/workflow.html#step-4-...`

File: docs/guide/input.rst [hunk 8]
```diff
@@ -0,0 +1,22 @@
+.. _input:
```
Comment (Chilipp) [inline docs/guide/input.rst]: this file should be included in some ``toctree`` directive.

File: arc3o/memls_functions_2D.py [hunk 9]
```diff
@@ -1360,43 +1718,61 @@ def layer(ri,s_i,ti,Ti,Tgnd,Tsky):
...
+    ev:	xarray.DataArray
+        emissivity v-pol
+
+    Notes
+    -----
+    This function is part of the original MEMLS developed by the Institute of Applied Physics,
+    University of Bern, Switzerland. A description of that model version can be found in :cite:`wiesmann98`
+    and :cite:`wiesmann99`. It was slightly modified to accomodate sea-ice layers by
+    `R.T. Tonboe <http://research.dmi.dk/staff/all-staff/rtt/>`_. It was translated to Python, adapted for multi-dimensional input and
+     further extended by `C. Burgard <http://www.github.com/ClimateClara>`_ to be used in ARC3O.
```
Comment (Chilipp) [inline arc3o/memls_functions_2D.py]:
```suggestion
    `R.T. Tonboe <http://research.dmi.dk/staff/all-staff/rtt/>`_. It was translated to Python, adapted for multi-dimensional input and
    further extended by `C. Burgard <http://www.github.com/ClimateClara>`_ to be used in ARC3O.
```

File: arc3o/profile_functions.py [hunk 10]
```diff
@@ -406,19 +487,32 @@ def build_density_profile(empty_dens_prof,temperature,salinity,winter_mask,layer
 
 def build_corrlen_profile(empty_corrlen_prof,prof_thick,fyi_mask,winter_mask,layer_amount):
     
+    """Build correlation length profiles.
+
+    This function builds correlation length profiles. Based on experiments by R.T. Tonboe, the correlation lengths are
+    defined as follows:
+        * First-year ice upper 20 cm: 0.25 mm
+        * First-year ice lower 20 cm: 0.35 mm
+        * Multiyear ice: 1.5 mm
```
Comment (Chilipp) [inline arc3o/profile_functions.py]:
```suggestion
    defined as follows:

    * First-year ice upper 20 cm: 0.25 mm
    * First-year ice lower 20 cm: 0.35 mm
    * Multiyear ice: 1.5 mm
```

File: docs/guide/step2.rst [hunk 11]
```diff
@@ -0,0 +1,21 @@
+Step 2: Prepare sea-ice profiles for cold conditions
```
Comment (Chilipp) [inline docs/guide/step2.rst]:
```suggestion
.. _step2:

Step 2: Prepare sea-ice profiles for cold conditions
```

File: docs/guide/step2.rst [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+described in :ref:`step1`, ARC3O adds the dimension *layer_number* to the lat-lon-time arrays. This way, it prepares two sets
+of profiles (ten ice layers and one snow layer), one set for snow-covered ice and one set for bare ice (see :func:`arc3o.profile_functions.create_profiles`). These profiles
+describe the layer:
+    * temperature :func:`arc3o.profile_functions.build_temp_profile`
+    * salinity :func:`arc3o.profile_functions.build_salinity_profile`
+    * thickness :func:`arc3o.profile_functions.build_thickness_profile`
+    * wetness :func:`arc3o.profile_functions.build_wetness_profile`
+    * density :func:`arc3o.profile_functions.build_density_profile`
+    * correlation length :func:`arc3o.profile_functions.build_corrlen_profile`
+    * type :func:`arc3o.profile_functions.build_sisn_profile`
```
Comment (Chilipp) [inline docs/guide/step2.rst]:
```suggestion
describe the layer:

* temperature :func:`arc3o.profile_functions.build_temp_profile`
* salinity :func:`arc3o.profile_functions.build_salinity_profile`
* thickness :func:`arc3o.profile_functions.build_thickness_profile`
* wetness :func:`arc3o.profile_functions.build_wetness_profile`
* density :func:`arc3o.profile_functions.build_density_profile`
* correlation length :func:`arc3o.profile_functions.build_corrlen_profile`
* type :func:`arc3o.profile_functions.build_sisn_profile`
```

**Review by Chilipp — CHANGES_REQUESTED** [review]
Hey @ClimateClara! Looks all good! Great job! I am so sorry, for taking so long :(

I just modified some of the formatting and fixed some issues that came while building the docs. Please have a look at the comments from this review for more details. 
I think they should be pretty much self-explanatory, but let me please know if anything is uncertain.

I also made a PR in https://github.com/ClimateClara/arc3o/pull/9 to add a `.gitignore` file to your repository. There is no need to keep the `_build` directory, particularly if you are using readthedocs to publish your docs.

**Comment by Chilipp** [comment]
note, I recommend that you add the suggested changes in one batch (_Add suggestion to batch_), then you have one commit per review, and not one commit per comment

File: setup.py (hunk 2 above)
Comment (ClimateClara) [reply setup.py]: I replaced this by inserting a 'requirements.txt'. Also I am not sure what to do about pathos.multiprocessing which needs to be installed from conda forge :/

File: readthedocs.yml (hunk 1 above)
Comment (ClimateClara) [reply readthedocs.yml]: It is not clear to me what this environment.yml should be? The one on which the package works or dependencies needed for the docs??

File: docs/guide/input.rst (hunk 8 above)
Comment (ClimateClara) [reply docs/guide/input.rst]: I have included the input info in the file howtorun.rst now so I think input.rst can be removed

File: setup.py (hunk 2 above)
Comment (Chilipp) [reply setup.py]: I see, well, this is not equivalent. If you want to publish your package on pypi.org, for instance, the dependencies need to be  defined in `setup.py` (or `setup.cfg`). having a `requirements.txt` on your repository root is just `good practice`

But you can skip all the built-in libraries here, so better use

```python
    install_requires=[
          'numpy',
          'xarray',
          'pandas',
          'tqdm',
          'pathos'
      ],
```

(note that the name of the package is [pathos](https://pypi.org/project/pathos/), not `pathos.multiprocessing`).

> Also I am not sure what to do about pathos.multiprocessing which needs to be installed from conda forge

At this point, there is no possibility to cover this. But you should nevertheless list it in `install_requires`. It's the same for `numpy` and `pandas`, which should better be installed via conda.

As long as you do not yet provide a conda package for arc3o, I'd recommend to add a note like 

_We recommend to install the dependencies via `conda install -c conda-forge pandas tqdm pathos` as they might not work well using `pip`_.

to your installation instructions. When the user then installs your package via `pip install arc3o`, it will realize that `pathos`, etc. is installed already and won't download it from pypi.org.

Comment (ClimateClara) [reply setup.py]: ok thanks, I'll do that then!

File: readthedocs.yml (hunk 1 above)
Comment (Chilipp) [reply readthedocs.yml]: this is an conda environment file ([example](https://github.com/psyplot/psyplot/blob/master/docs/environment.yml)) that tells readthedocs.org about the dependencies you need to build the docs (see [here](https://docs.readthedocs.io/en/stable/guides/conda.html)). You should list here additional sphinx extensions, or dependencies of your package that you do not want to have installed via `pip` (`pathos` and `pandas` for instance).

Note that, from my experience, readthedocs does not work with the conda-forge channel, as it eats too much memory. I therefore took necessary packages from the conda-forge channel and added them to my own channel which I am then using in the `environment.yml` file.

File: docs/start/installation.rst (hunk 3 above)
Comment (ClimateClara) [reply docs/start/installation.rst]: inserted it, thanks, I was a bit unsure about how this works!

File: readthedocs.yml (hunk 1 above)
Comment (ClimateClara) [reply readthedocs.yml]: Hmm ok thanks. Still unclear to me what you mean by your "own channel". What is this channel? Is this in github or somewhere else? Sorry if this is a stupid question :/

Comment (ClimateClara) [reply readthedocs.yml]: but if I do not mention anything of pandas or pathos in the docs, do I still need this? I am a little confused about the purpose of this environment

Comment (Chilipp) [reply readthedocs.yml]: > What is this channel? Is this in github or somewhere else?

it's an anaconda channel. You can register there and publish your own packages under your own channel (here is mine: https://anaconda.org/chilipp/)

> but if I do not mention anything of pandas or pathos in the docs, do I still need this?

You are using the directives of `sphinx.ext.autodoc` (`.. automodule::` for instance). This will import your package when building the docs. If `pathos` is not needed for building the docs (i.e. it is not imported on a module level), you can also exclude it

Comment (Chilipp) [reply readthedocs.yml]: an alternative is, to import pathos only, if you are not building the docs on readthedocs. In your `core_functions.py` for instance, you could change the `pathos` import into something like

```python
import os
if not os.getenv('READTHEDOCS'):
    from pathos.multiprocessing import ProcessingPool as Pool
```

this is not a super nice solution, but it would be the easiest

Comment (ClimateClara) [reply readthedocs.yml]: then let's do it this way for the moment and hope it works. I have now added an environment.yml file, inspired from the conda environment I used when working on the docs. I am a bit confused because autodocsumm is not there. Did I do something wrong?

Comment (Chilipp) [reply readthedocs.yml]: > I am a bit confused because autodocsumm is not there

not sure what you mean. Do you mean, it's not available through the `defaults` conda channel? You could list it to be installed via pip in your `environment.yml`, see https://github.com/psyplot/psyplot/blob/152bf556cc68142060ba06e91905460d74250f23/docs/environment.yml#L19 for instance

Comment (ClimateClara) [reply readthedocs.yml]: Sorry, that was probably not clear. I meant that I created the environment directly from my working conda env (with  conda env export > environment.yml) and the resulting environment.yml does not contain autodocsumm. I found this surprising. Anyway, I'll add it then

Comment (Chilipp) [reply readthedocs.yml]: ah, I see :sweat_smile: then we can mark this as resolved, I suppose :smiley:

**Comment by Chilipp** [comment]
then we just need to resolve the merge conflicts, and we're done here I suppose. Or do you want to setup readthedocs with this branch?

**Comment by ClimateClara** [comment]
cool! Yes let's merge this first. How much extra things need to be done from the github side for the upload to readthedocs? Do we need a new branch for that?

**Comment by Chilipp** [comment]
> How much extra things need to be done from the github side for the upload to readthedocs? Do we need a new branch for that?

ideally it should work out-of-the-box with what you have done here :wink:

**Review by Chilipp — APPROVED** [review]

**Comment by ClimateClara** [comment]
cool I'll try it then :)
